pytorch_lightning/core/optimizer.py

```diff
@@ -59,8 +59,10 @@ def __init__(self,
 
         self._trainer = None
         self._optimizer = optimizer
+        self._optimizer_idx = None
         self._accumulate_grad_batches = accumulate_grad_batches
         self._use_accumulate_grad_batches_from_trainer = accumulate_grad_batches is None
+        self._in_foreach_worker = False
```

Comment:
Could we put this variable into the ddp_plugin where it could be set? This way the optimizer is cleaner